src/main/kotlin/com/openlattice/assembler/AssemblerConnectionManager.kt

```diff
@@ -614,85 +656,59 @@ class AssemblerConnectionManager(
                 logger.info("Created user mapping. ")
                 statement.execute(importProductionViewsSchemaSql(setOf()))
                 statement.execute(
-                        importPublicSchemaSql(setOf(EDGES.name, PROPERTY_TYPES.name, ENTITY_TYPES.name, ENTITY_SETS.name))
+                        importPublicSchemaSql(PUBLIC_TABLES)
                 )
                 logger.info("Imported foreign schema")
             }
         }
     }
 
-    fun createProductionForeignSchemaOfEntitySetIfNotExists(organizationId: UUID, entitySetId: UUID) {
-        val dbname = PostgresDatabases.buildOrganizationDatabaseName(organizationId)
-        connect(dbname).use { datasource ->
-            createProductionForeignSchemaOfEntitySetIfNotExists(datasource, entitySetId)
+    private fun updatePublicTables(datasource: HikariDataSource, tables: Set<String>) {
+        datasource.connection.use { connection ->
+            connection.createStatement().use { stmt ->
+                tables.forEach {
+                    stmt.execute(dropProductionViewSchemaSql(it))
+                }
+                stmt.execute(importPublicSchemaSql(tables))
+            }
         }
     }
 
-    private fun createProductionForeignSchemaOfEntitySetIfNotExists(dataSource: HikariDataSource, entitySetId: UUID) {
-        dataSource.connection.use { conn ->
-            // check if table exists and import if not
-            if (!checkIfProductionViewsSchemaExists(conn, entitySetId)) {
-                conn.createStatement().use { stmt ->
-                    stmt.execute(importProductionViewsSchemaSql(setOf(DataTables.quote(entitySetId.toString()))))
-                    // (re)-import entity_sets
-                    updatePublicTables(stmt, setOf(PostgresTable.ENTITY_SETS.name))
+    private fun updateProductionViewTables(datasource: HikariDataSource, tables: Set<String>) {
```

Comment:
are these two functions doing the same thing? (`updatePublicTables` and `updateProductionViewTables`)

Reply:
Not exactly, one is importing foreign tables from "public"(PUBLIC_SCHEMA) schema and the other one from "olviews" (PRODUCTION_VIEWS_SCHEMA) schema, but both are importing it into "prod" (PRODUCTION_FOREIGN_SCHEMA) schema in org database.
But I will rename "dropProductionViewSchemaSql", because it's misleading, it actually drops tables from PRODUCTION_FOREIGN_SCHEMA and not PRODUCTION_VIEWS_SCHEMA